app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java

```diff
@@ -18,23 +20,24 @@
       R.layout.activity_request_state_declined
   };
 
-  private SelectShopperActivity.SelectShopperAdapter.ShopRequest request;
+  private OutgoingRequest request;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     request = getIntent().getParcelableExtra(EXTRAS_REQUEST_KEY);
-    setContentView(LAYOUTS[request.status.ordinal()]);
+    setContentView(LAYOUTS[request.getStatus().ordinal()]);
     ButterKnife.inject(this);
   }
 
   @Optional
   @OnClick(R.id.request_state_cancel_button)
   public void onCancelRequest(View view) {
+    request.setStatus(BaseRequest.RequestStatus.CANCELED);
     // TODO: Cancel request in server
-    request.shopper = null;
+    OutgoingRequest newRequest = new OutgoingRequest(request.getShoppingList());
     Intent intent = new Intent(this, SelectShopperActivity.class);
-    intent.putExtra(SelectShopperActivity.EXTRAS_REQUEST_KEY, request);
+    intent.putExtra(SelectShopperActivity.EXTRAS_REQUEST_KEY, newRequest);
```

Comment:
The "session" is hard to define in android.
If you go to the home screen for a bit and then come back you might need to reload everything.

I think each activity should only deal with the data that it actually needs. It makes it much easier to change things later and also makes it easier to reuse activities (if they do a very well-defined thing, like editing a shopping list).
I don't think we'll really have a stitch-together stage, since you will eventually pass the data that the API needs to some API class, and it'll have to return a new `Request` object anyway, with the correct state of all the fields.
So you either slowly assemble a `Request` by coupling all the activities to that object, then take it apart to send it over the network, then create a new one, or just hold the relevant pieces when you actually need them.